# AAI614: Data Science & its Applications

*Notebook 2.5: Practice with Parquet and File Types*

<a href="https://colab.research.google.com/github/harmanani/AAI614/blob/main/Week%202/Notebook2.5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd
import time
import ssl

ssl._create_default_https_context = ssl._create_unverified_context

In [ ]:
class Timer:
    def __enter__(self):
        self.start = time.perf_counter()
        return self

    def __exit__(self, *args):
        self.end = time.perf_counter()
        self.interval = self.end - self.start

##### Read the Parqeut file and time it

In [14]:
with Timer() as t_pd:
    df = pd.read_parquet('https://raw.githubusercontent.com/harmanani/AAI614/main/Week%202/niaaa-report.parquet')
print(t_pd.interval)

0.05334210699993491


#### Read the CSV file and time it

In [15]:
with Timer() as t_pd:
    df = pd.read_csv('https://raw.githubusercontent.com/harmanani/AAI614/main/Week%202/niaaa-report.csv')
print(t_pd.interval)

0.06411816899981204


#### Read the ZIP file and time it

In [16]:
import zipfile
with Timer() as t_pd:
    df = pd.read_csv('https://raw.githubusercontent.com/harmanani/AAI614/main/Week%202/niaaa-report.zip', compression="zip")
print(t_pd.interval)

0.0571223749998353


In [18]:
with Timer() as t:
    df = pd.read_csv("https://raw.githubusercontent.com/harmanani/AAI614/refs/heads/main/Week%202/nfl.csv")
print(t.interval)

0.06048085499969602


In this lab, I measured how long it took to load the same dataset using three different file formats: Parquet, CSV, and ZIP, with the provided Timer class. The approximate loading times were:

Parquet: 0.0533 seconds
CSV: 0.0641 seconds
ZIP: 0.0571 seconds
I also loaded another CSV file (nfl.csv), which took about 0.0605 seconds.
Overall, the loading times were very similar. In this run, Parquet was the fastest, followed by ZIP, and then CSV. However, these small differences can change from run to run because of factors like network conditions, file size, and caching. The experiment shows that the file format can influence how quickly data is read, and that performance depends on the specific dataset and environment. Parquet is generally a good choice for larger datasets because it is optimized for efficient storage and fast processing.